yt_dlp/extractor/ard.py

```diff
@@ -360,11 +361,29 @@ def _extract_episode_info(self, title):
     def _real_extract(self, url):
         display_id = self._match_id(url)
 
-        page_data = self._download_json(
-            f'https://api.ardmediathek.de/page-gateway/pages/ard/item/{display_id}', display_id, query={
-                'embedded': 'false',
-                'mcV6': 'true',
-            })
+        token = self._download_json('https://sso.ardmediathek.de/sso/token', display_id)
+
+        id_token = traverse_obj(token, 'idToken')
+
+        if id_token:
+            jwt_token_decoded = jwt_decode_hs256(id_token)
+            user_id = traverse_obj(jwt_token_decoded, 'user_id', 'sub')
+
+            page_data = self._download_json(
+                f'https://api.ardmediathek.de/page-gateway/pages/ard/item/{display_id}', display_id, query={
+                    'embedded': 'false',
+                    'mcV6': 'true',
+                    'userId': user_id
+                }, headers= {
+                    'x-authorization': f'Bearer {id_token}'
+                })
+        else:
+            page_data = self._download_json(
+                f'https://api.ardmediathek.de/page-gateway/pages/ard/item/{display_id}', display_id, query={
+                    'embedded': 'false',
+                    'mcV6': 'true'
+                })
```

Comment:
I would like to avoid having duplication here, maybe by instead writing to the query instead?